nemo_curator/utils/distributed_utils.py

```diff
@@ -273,6 +273,18 @@ def _set_torch_to_use_rmm():
     torch.cuda.memory.change_current_allocator(rmm_torch_allocator)
 
 
+def _resolve_filename_col(filename: Union[bool, str]) -> Union[str, bool]:
```

Comment:
I like this a lot, thanks! Was wondering if other files from https://github.com/NVIDIA/NeMo-Curator/pull/449 (like doc_dataset.py) need to be updated as well?

Reply:
Oops looks like I forgot all the APIs where those changes needed to be made. Added those too now thanks!

Reply:
I still think you could be missing some?

- https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/download/doc_builder.py
- https://github.com/NVIDIA/NeMo-Curator/blob/main/docs/user-guide/documentdataset.rst
- https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/utils/file_utils.py

And writing logic such as in https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/utils/distributed_utils.py.